src/backend/src/services/projects.services.ts

```diff
@@ -811,4 +811,39 @@ export default class ProjectsService {
 
     return newMaterialType;
   }
+
+  /**
+   * Assign a material on a project to a different assembly
+   * @param submitter the submitter
+   * @param materialId the material that will be moved
+   * @param assemblyId the assembly to change the material to
+   * @throws if the submitter does not have the relevant positions
+   * @returns the updated material
+   */
+  static async assignMaterialAssembly(submitter: User, materialId: string, assemblyId: string) {
+    const material = await prisma.material.findUnique({ where: { materialId } });
+    if (!material) throw new NotFoundException('Material', materialId);
+
+    const assembly = await prisma.assembly.findUnique({
```

Comment:
ok so there's some complexity here that I didn't think ab when I made the ticket. If assemblyId is undefined, we wanna unassign the material from whatever assembly it's currently assigned to (bc/ materials can be assigned to no assembly). 

Also if there is an assemblyId submitted, we should probably check that its wbsElement is the same as the wbsElement on the material that was submitted